lib/mutant/repository/diff.rb

```diff
@@ -22,12 +22,16 @@ class Error < RuntimeError; end
       # @raise [RepositoryError]
       #   when git command failed
       def touches?(path, line_range)
-        touched_paths
-          .from_right { |message| fail Error, message }
-          .fetch(path) { return false }
+        touched_path(path) { return false }
           .touches?(line_range)
       end
 
+      def touches_path?(path)
```

Comment:
@dgollahon This is a very valid interface addition to `Repository::Diff` I'll extract this to a separate PR to base my universal solution for reducing to be parsed methods down to a smaller amount.